src/main/java/IO/DbReader.java

```diff
@@ -0,0 +1,53 @@
+
+package IO;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+
+import data.Database;
+
+/* DbReader class accesses blob information to read its contents. */
+public class DbReader {
+
+    private Database database;
+    private String runId;
+    private String type;
+    private InputStream in;
+
+    /* Comments out extension variable bc may not need it
+     private static final String extension = ".bin"; . */
+    private String fileName;
+    
+    /* Makes a reader object. */
+    public DbReader(Database database, String runId, String type) {
+        this.database = database;
+        this.runId = runId;
+        this.type = type;
+        fileName = database.findName(this.runId, this.type);
```

Comment:
You don't need the fileName, since it's the implementation detail of db, the user doesn't need to know it, let db provide api to readData(runId, type)